docs/remote-attestation.md

```diff
@@ -1,212 +1,111 @@
-     includes a hash of the **Trusted Runtime**’s _Signing_ public key
-   - Corresponding **TEE Provider**’s certificate that is signed by the **TEE
-     Provider**’s _Root_ key
-5. **Client** validates `ServerIdentity`
-   - If the corresponding `AttestationReport` is not valid, then the **Client**
-     closes the connection and aborts the protocol
-6. **Client** sends `ClientIdentity` to the **Trusted Runtime** which contains:
-   - **Client**’s _Ephemeral_ public key
-   - _Transcript_: [SHA-256](https://datatracker.ietf.org/doc/html/rfc6234) hash
-     of the concatenated `ClientHello`, `ServerIdentity` and current
-     `ClientIdentity` (excluding the _Transcript_) signed with the **Client**'s
-     _Signing_ private key
-   - **Client**’s _Signing_ public key
-7. **Client** and **Trusted Runtime** use both _Ephemeral_ public keys to create
-   an [X25519](https://datatracker.ietf.org/doc/html/rfc7748) _Shared Secret_
-8. **Client** and **Trusted Runtime** derive _Session_ keys from the _Shared
-   Secret_ and both _Ephemeral_ public keys using a _Key Derivation Function_
-   ([HKDF](https://datatracker.ietf.org/doc/html/rfc5869))
-   - Each side generates 2
-     [AES-256-GCM](https://datatracker.ietf.org/doc/html/rfc5288) _Session_
-     keys:
-     - Server Session Key
-     - Client Session Key
-   - **Client** and **Trusted Runtime** use Authenticated Encryption/Decryption
-     ([AEAD](https://en.wikipedia.org/wiki/Authenticated_encryption)) for
-     communication
-
-### Post-handshake
-
-**Client** and **Trusted Runtime** use generated _Session_ keys to exchange
-data.
+**Remote Attestation** allows a client to remotely verify the TEE hardware
+authenticity and the identity of the code running inside the TEE.
+
+If the code of the application is **reproducibly buildable**, the client can
+also check that the TEE is running the application that is expected to run.
+
+The Remote Attestation process is based on an **Attestation Report**, a data
+structure signed by the TEE platform and containing information identifying the
+code that is running inside the TEE. This report can be checked to confirm that
+it is signed by the **TEE Provider** (e.g. AMD or Intel), which results in
+evidence that the code is running on a genuine TEE platform.
 
 ## Workflow Diagram
 
 ```mermaid
 sequenceDiagram
-  participant Client
-  participant Untrusted Launcher
-  participant Trusted Runtime
-  participant TEE Platform
 
-  Note over Untrusted Launcher,TEE Platform: Load the Trusted Runtime
-
-  Untrusted Launcher->>TEE Platform: Trusted Runtime Memory Contents
-  TEE Platform -->> TEE Platform: Store a measurement of the Trusted Runtime
-  TEE Platform ->> Trusted Runtime: Start the Trusted Runtime in a secure VM
+autonumber
+
+participant C as Client
+participant U as Untrusted Launcher
+participant T as Trusted Enclave
+participant S as Secure Processor
+
+Note over U,S: Boot Trusted Enclave
+
+U->>S: Trusted Enclave Binary + firmware
```

Comment:
This is not 100% accurate. This goes via the VMM/hypervisor, and most of the work is done there. The secure processor is only involved in encrypting and measuring the pages after the hypervisor loaded the firmware and enclave binary into the guest memory.